src/test/groovy/com/getkeepsafe/dexcount/DexMethodCountPluginSpec.groovy

```diff
@@ -153,4 +158,118 @@ final class DexMethodCountPluginSpec extends Specification {
     project.tasks.getByName("countFlavor2Flavor4DebugDexMethods")
     project.tasks.getByName("countFlavor2Flavor4ReleaseDexMethods")
   }
+
+  def "android apk report example"() {
+    given:
+    def apkFile = tempFolder.newFile("tiniest-smallest-app.apk")
+    def apkResource = getClass().getResourceAsStream("/tiniest-smallest-app.apk")
+    apkResource.withStream { input ->
+      IOUtil.drainToFile(input, apkFile)
+    }
+
+    project.apply plugin: "com.android.application"
+    project.apply plugin: "com.getkeepsafe.dexcount"
+    project.android {
+      compileSdkVersion COMPILE_SDK_VERSION
+      buildToolsVersion BUILD_TOOLS_VERSION
+
+      defaultConfig {
+        applicationId APPLICATION_ID
+      }
+    }
+
+    when:
+    project.evaluate()
+
+    // Override APK file
+    DexMethodCountTask task = project.tasks.getByName("countDebugDexMethods") as DexMethodCountTask
+    task.apkOrDex = Mock(BaseVariantOutput)
+    task.apkOrDex.outputFile >> apkFile
+    task.execute()
+
+    then:
+    // debug.csv - CSV
+    def actualOutputFile = task.outputFile.absoluteFile.text
+    def expectedOutputFile = "methods  fields   package/class name\n" +
+            "6        0        android\n" +
+            "2        0        android.app\n" +
+            "4        0        android.widget\n" +
+            "3        0        b\n" +
+            "3        0        b.a\n"
+    // debug.txt - TXT
+    def actualSummaryFile = task.summaryFile.absoluteFile.text
+    def expectedSummaryFile = "methods,fields\n" +
+            "9,0\n"
+    // debugChart/data.js - JSON
+    def actualChartDir = new File(task.chartDir, "data.js").text
+    def expectedChartDir = "var data = {\n" +
```

Comment:
It would be nicer to have this expected output in some resource file for future updates